### PR TITLE
Rel notes for storage 4.18

### PR DESCRIPTION
**Pull request by lpettyjo**
Version(s): 4.18

Issue:

Link to docs preview:
**Feature blurbs:** https://86629--ocpdocs-pr.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-18-release-notes.html#ocp-release-notes-storage_release-notes

**Tech Preview status**: https://86629--ocpdocs-pr.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-18-release-notes.html#ocp-4-18-technology-preview-tables_release-notes

**Removed features:** https://86629--ocpdocs-pr.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-18-release-notes.html#ocp-4-18-removed-features_release-notes

QE review:
- [x] QE has approved this change.

Additional information:

**PTAL**: @gcharot @jsafrane @dobsonj @gnufied @dfajmon @vr4manta @RomanBednar @mpatlasov @gouthampacha @skopacz1 @rlobillo @Phaow @ropatil010 @wsun1 @chao007 @duanwei33 @radeore

### CODE REVIEW

File: release_notes/ocp-4-18-release-notes.adoc [hunk 1]
```diff
@@ -474,9 +473,72 @@ The terminology for the `ip_whitelist` and `ip_blacklist` annotations have been
...
+
+[id="ocp-4-18-release-notes-storage-secret-csi-driver_{context}"]
+==== Secret Store CSI Driver Operator is generally available
+The Secrets Store Container Storage Interface (CSI) Driver Operator, `secrets-store.csi.k8s.io,` allows {product-title} to mount multiple secrets, keys, and certificates stored in enterprise-grade external secrets stores into pods as an inline ephemeral volume. The Secrets Store CSI Driver Operator communicates with the provider using gRPC to fetch the mount contents from the specified external secrets store. After the volume is attached, the data in it is mounted into the container’s file system. The Secrets Store CSI Driver Operator was available in {product-title} 4.14 as a Technology Preview feature. {product-title} 4.18 introduces this feature as generally available. For more information about the Secrets Store CSI driver, see xref:../storage/container_storage_interface/persistent-storage-csi-secrets-store.adoc#persistent-storage-csi-secrets-store[Secrets Store CSI Driver Operator].
```
Comment (xenolinux) [inline release_notes/ocp-4-18-release-notes.adoc:486]:
```suggestion
The Secrets Store Container Storage Interface (CSI) Driver Operator, `secrets-store.csi.k8s.io`, allows {product-title} to mount multiple secrets, keys, and certificates stored in enterprise-grade external secrets stores into pods as an inline ephemeral volume. The Secrets Store CSI Driver Operator communicates with the provider using gRPC to fetch the mount contents from the specified external secrets store. After the volume is attached, the data in it is mounted into the container’s file system. The Secrets Store CSI Driver Operator was available in {product-title} 4.14 as a Technology Preview feature. {product-title} 4.18 introduces this feature as generally available. For more information about the Secrets Store CSI driver, see xref:../storage/container_storage_interface/persistent-storage-csi-secrets-store.adoc#persistent-storage-csi-secrets-store[Secrets Store CSI Driver Operator].
```

-  A tiny nit to adjust the comma. 
-  You can consider dividing this into paragraphs.

File: release_notes/ocp-4-18-release-notes.adoc [hunk 2]
```diff
@@ -474,9 +473,72 @@ The terminology for the `ip_whitelist` and `ip_blacklist` annotations have been
...
+
+{product-title} supports volume group snapshots by default. However, the only storage type in {product-title} that supports volume group snapshots is {rh-storage}.
+
+//For more information about CSI volume group snapshots, see xref:/../storage/container_storage_interface/persistent-storage-csi-group-snapshots.adoc#persistent-storage-csi-group-snapshots[CSI volume group snapshots].
+
+[id="ocp-4-18-release-notes-storage-c3-n4-instance-types-hyperdisk_{context}"]
+==== GCP PD CSI driver supports the C3 instance type for bare metal and N4 machine series is generally available
+The Google Cloud Platform Persistent Disk(GCP PD) Container Storage Interface (CSI) driver supports the C3 instance type for bare metal and N4 machine series. The C3 instance type and N4 machine series support the hyperdisk-balanced disks.
```
Comment (xenolinux) [inline release_notes/ocp-4-18-release-notes.adoc:528]:
```suggestion
The Google Cloud Platform Persistent Disk (GCP PD) Container Storage Interface (CSI) driver supports the C3 instance type for bare metal and N4 machine series. The C3 instance type and N4 machine series support the hyperdisk-balanced disks.
```

File: release_notes/ocp-4-18-release-notes.adoc [hunk 3]
```diff
@@ -474,9 +473,72 @@ The terminology for the `ip_whitelist` and `ip_blacklist` annotations have been
...
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-multi-vcenter-support-overview_persistent-storage-csi-vsphere[Multiple vCenter support for vSphere CSI] and xref:../installing/installing_vsphere/installation-config-parameters-vsphere.adoc[Installation configuration parameters for vSphere].
+
+[id="ocp-4-18-release-notes-storage-always-honor-reclaim-policy_{context}"]
+==== Always apply persistent volume reclaim policy (Technical Preview)
+Prior to {product-title} 4.18, the persistent volume (PV) reclaim policy was not always honored.
+
+For a bound PV-persistent volume pair (PVC) pair, the ordering of PV-PVC deletion determined whether the PV delete reclaim policy was applied or not. The PV applied the reclaim policy if the PVC was deleted prior to deleting the PV. However, if the PV was deleted prior to deleting the PVC, then the reclaim policy was not applied. As a result of that behavior, the associated storage asset in the external infrastructure was not removed.
```
Comment (xenolinux) [inline release_notes/ocp-4-18-release-notes.adoc:506]:
```suggestion
For a bound PV-persistent volume claim (PVC) pair, the ordering of PV-PVC deletion determined whether the PV delete reclaim policy was applied or not. The PV applied the reclaim policy if the PVC was deleted prior to deleting the PV. However, if the PV was deleted prior to deleting the PVC, then the reclaim policy was not applied. As a result of that behavior, the associated storage asset in the external infrastructure was not removed.
```

I believe it should be "claim"

**Review by xenolinux — COMMENTED** [review]
Tiny nits -- Otherwise LGTM!!

File: release_notes/ocp-4-18-release-notes.adoc [hunk 4]
```diff
@@ -821,6 +888,10 @@ In the following tables, features are marked with the following statuses:
 [id="ocp-4-18-removed-features_{context}"]
 === Removed features
 
+[id="ocp-4-18-removals-storage_{context}"]
+==== The Shared Resource CSI Driver is removed
+The Shared Resource CSI Driver feature is now removed from {product-title} 4.18. It was deprecated in {product-title} 4.17. This feature is now generally available in Builds for Red Hat OpenShift 1.1. To use this feature, ensure you are using Builds for Red Hat OpenShift 1.1 or later.
```
Comment (Phaow) [inline release_notes/ocp-4-18-release-notes.adoc:893]:
```suggestion
The Shared Resource CSI Driver feature was deprecated in {product-title} 4.17 and is now removed from {product-title} 4.18. 
```
Maybe better.

File: release_notes/ocp-4-18-release-notes.adoc [hunk 5]
```diff
@@ -474,9 +473,72 @@ The terminology for the `ip_whitelist` and `ip_blacklist` annotations have been
...
+
+For more information about using the persistent volume last phase transition time parameter, see xref:../storage/understanding-persistent-storage.adoc#persistent-volumes_understanding-persistent-storage[Persistent volumes].
+
+[id="ocp-4-18-release-notes-storage-volume-multi-vcenter-support_{context}"]
+==== Multiple vCenter support for vSphere CSI is generally available
+{product-title} 4.17 introduced the ability to deploy {product-title} across multiple vSphere clusters (vCenters) as a Technology Preview feature. In {product-title} 4.18, Multiple vCenter support is now generally available.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-multi-vcenter-support-overview_persistent-storage-csi-vsphere[Multiple vCenter support for vSphere CSI] and xref:../installing/installing_vsphere/installation-config-parameters-vsphere.adoc[Installation configuration parameters for vSphere].
```
Comment (duanwei33) [inline release_notes/ocp-4-18-release-notes.adoc:500]: It's not the scope of this RN PR,  but let's ensure we update the reference `Multiple vCenter support for vSphere CSI` to remove the TP.

File: release_notes/ocp-4-18-release-notes.adoc [hunk 6]
```diff
@@ -474,9 +473,72 @@ The terminology for the `ip_whitelist` and `ip_blacklist` annotations have been
...
+For more information, see xref:../storage/understanding-persistent-storage.adoc#reclaiming_understanding-persistent-storage[Reclaim policy for persistent volumes].
+
+[id="ocp-4-18-release-notes-storage-lso-cleanup_{context}"]
+==== Improved ability to easily remove LVs or LVSs is generally available
+{product-title} 4.18 improves the ability to remove Local Volumes (LVs) and Local Volume Sets (LVSs) by automatically removing artifacts, thus reducing the number of steps required.
+
+For more information, see xref:../storage/persistent_storage/persistent_storage_local/persistent-storage-local.adoc#local-removing-device_persistent-storage-local[Removing a local volume or local volume set].
+
+[id="ocp-4-18-release-notes-storage-volume-group-snapshots_{context}"]
+==== CSI volume group snapshots is generally available
```
Comment (duanwei33) [inline release_notes/ocp-4-18-release-notes.adoc:519]: The content looks good, but please note the code is still not ready now and we need to recheck this part before release.

File: release_notes/ocp-4-18-release-notes.adoc [hunk 7]
```diff
@@ -474,9 +473,72 @@ The terminology for the `ip_whitelist` and `ip_blacklist` annotations have been
...
 Previously, the `thinPoolConfig.overprovisionRatio` field in the `LVMCluster` custom resource (CR) could be configured only during the creation of the `LVMCluster` CR. With this release, you can now update the `thinPoolConfig.overprovisionRatio` field even after creating the `LVMCluster` CR.
 
+[id="ocp-4-18-release-notes-storage-cifs-smb-csi-driver_{context}"]
+==== Persistent storage using CIFS/SMB CSI Driver Operator is generally available
+{product-title} is capable of provisioning persistent volumes (PVs) with a Container Storage Interface (CSI) driver for the Common Internet File System (CIFS) dialect/Server Message Block (SMB) protocol. The CIFS/SMB CSI Driver Operator that manages this driver was introduced in {product-title} 4.16 with Technology Preview status. In {product-title} 4.18, it is now generally available.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-smb-cifs.adoc#persistent-storage-csi-smb-cifs[CIFS/SMB CSI Driver Operator].
+
+[id="ocp-4-18-release-notes-storage-secret-csi-driver_{context}"]
+==== Secret Store CSI Driver Operator is generally available
```
Comment (ropatil010) [inline release_notes/ocp-4-18-release-notes.adoc:485]: Content looks good to me for Secret Store part. 
Double confirm "to mount multiple secrets, keys, and certificates stored" => "to mount multiple secrets, keys and certificates stored" would be better?

File: release_notes/ocp-4-18-release-notes.adoc [hunk 8]
```diff
@@ -474,9 +473,72 @@ The terminology for the `ip_whitelist` and `ip_blacklist` annotations have been
...
+
+// For information about using the Secrets Store CSI Driver Operator to mount secrets from an external secrets store to a CSI volume, see xref:../nodes/index#nodes-pods-secrets-store[Providing sensitive data to pods by using an external secrets store].
+
+[id="ocp-4-18-release-notes-storage-pv-last-phase-transition-time_{context}"]
+==== Persistent volume last phase transition time parameter is generally available
+{product-title} 4.16 introduced a new parameter, `LastPhaseTransitionTime`, which has a timestamp that is updated every time a persistent volume (PV) transitions to a different phase (`pv.Status.Phase`). For {product-title} 4.18, this feature is generally available.
```
Comment (radeore) [inline release_notes/ocp-4-18-release-notes.adoc:492]: Content LGTM

File: release_notes/ocp-4-18-release-notes.adoc [hunk 9]
```diff
@@ -474,9 +473,72 @@ The terminology for the `ip_whitelist` and `ip_blacklist` annotations have been
...
+
+For more information about using the persistent volume last phase transition time parameter, see xref:../storage/understanding-persistent-storage.adoc#persistent-volumes_understanding-persistent-storage[Persistent volumes].
```
Comment (radeore) [inline release_notes/ocp-4-18-release-notes.adoc:494]: This reference takes to PV docs, but doesn't have details about "last phase transition time parameter"

File: release_notes/ocp-4-18-release-notes.adoc [hunk 10]
```diff
@@ -474,9 +473,72 @@ The terminology for the `ip_whitelist` and `ip_blacklist` annotations have been
...
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-multi-vcenter-support-overview_persistent-storage-csi-vsphere[Multiple vCenter support for vSphere CSI] and xref:../installing/installing_vsphere/installation-config-parameters-vsphere.adoc[Installation configuration parameters for vSphere].
+
+[id="ocp-4-18-release-notes-storage-always-honor-reclaim-policy_{context}"]
+==== Always apply persistent volume reclaim policy (Technical Preview)
+Prior to {product-title} 4.18, the persistent volume (PV) reclaim policy was not always honored.
+
+For a bound PV-persistent volume claim (PVC) pair, the ordering of PV-PVC deletion determined whether the PV delete reclaim policy was applied or not. The PV applied the reclaim policy if the PVC was deleted prior to deleting the PV. However, if the PV was deleted prior to deleting the PVC, then the reclaim policy was not applied. As a result of that behavior, the associated storage asset in the external infrastructure was not removed.
```
Comment (radeore) [inline release_notes/ocp-4-18-release-notes.adoc:506]:
```suggestion
For a bound persistent volume (PV)-persistent volume claim (PVC) pair, the ordering of PV-PVC deletion determined whether the PV delete reclaim policy was applied or not. The PV applied the reclaim policy if the PVC was deleted prior to deleting the PV. However, if the PV was deleted prior to deleting the PVC, then the reclaim policy was not applied. As a result of that behavior, the associated storage asset in the external infrastructure was not removed.
```

File: release_notes/ocp-4-18-release-notes.adoc [hunk 11]
```diff
@@ -474,9 +473,72 @@ The terminology for the `ip_whitelist` and `ip_blacklist` annotations have been
...
+For more information about using the persistent volume last phase transition time parameter, see xref:../storage/understanding-persistent-storage.adoc#persistent-volumes_understanding-persistent-storage[Persistent volumes].
+
+[id="ocp-4-18-release-notes-storage-volume-multi-vcenter-support_{context}"]
+==== Multiple vCenter support for vSphere CSI is generally available
+{product-title} 4.17 introduced the ability to deploy {product-title} across multiple vSphere clusters (vCenters) as a Technology Preview feature. In {product-title} 4.18, Multiple vCenter support is now generally available.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-multi-vcenter-support-overview_persistent-storage-csi-vsphere[Multiple vCenter support for vSphere CSI] and xref:../installing/installing_vsphere/installation-config-parameters-vsphere.adoc[Installation configuration parameters for vSphere].
+
+[id="ocp-4-18-release-notes-storage-always-honor-reclaim-policy_{context}"]
+==== Always apply persistent volume reclaim policy (Technical Preview)
```
Comment (radeore) [inline release_notes/ocp-4-18-release-notes.adoc:503]:
```suggestion
==== Always honor persistent volume reclaim policy (Technical Preview)
```

Comment (radeore) [reply release_notes/ocp-4-18-release-notes.adoc:503]: In case to keep title consistent with Upstream feature

File: release_notes/ocp-4-18-release-notes.adoc [hunk 12]
```diff
@@ -474,9 +473,72 @@ The terminology for the `ip_whitelist` and `ip_blacklist` annotations have been
...
+Prior to {product-title} 4.18, the persistent volume (PV) reclaim policy was not always honored.
+
+For a bound PV-persistent volume claim (PVC) pair, the ordering of PV-PVC deletion determined whether the PV delete reclaim policy was applied or not. The PV applied the reclaim policy if the PVC was deleted prior to deleting the PV. However, if the PV was deleted prior to deleting the PVC, then the reclaim policy was not applied. As a result of that behavior, the associated storage asset in the external infrastructure was not removed.
+
+With {product-title} 4.18, the PV reclaim policy is consistently always applied. This feature has Technical Preview status.
```
Comment (radeore) [inline release_notes/ocp-4-18-release-notes.adoc:508]: Should we have this feature listed under "Storage TechPreview features" table? cc. @duanwei33

File: release_notes/ocp-4-18-release-notes.adoc (hunk 4 above)
Comment (lpettyjo) [reply release_notes/ocp-4-18-release-notes.adoc:893]: Done.

File: release_notes/ocp-4-18-release-notes.adoc (hunk 7 above)
Comment (lpettyjo) [reply release_notes/ocp-4-18-release-notes.adoc:485]: We generally use the serial (aka Oxford comma).

File: release_notes/ocp-4-18-release-notes.adoc (hunk 9 above)
Comment (lpettyjo) [reply release_notes/ocp-4-18-release-notes.adoc:494]: @radeore Correct. Since the PR for the "last phase transition time parameter" hasn't merged yet, you can't see the content, but the link is to the correct file.

File: release_notes/ocp-4-18-release-notes.adoc (hunk 5 above)
Comment (lpettyjo) [reply release_notes/ocp-4-18-release-notes.adoc:500]: @duanwei33 Yes. That is covered in this PR https://github.com/openshift/openshift-docs/pull/85971.

File: release_notes/ocp-4-18-release-notes.adoc (hunk 10 above)
Comment (lpettyjo) [reply release_notes/ocp-4-18-release-notes.adoc:506]: @radeore It's awkward, but the PV acronym is define in the sentence above this, but PVC is not, so it needs to be defined here. I've removed the hyphen to make this a little less jarring (I hope).

File: release_notes/ocp-4-18-release-notes.adoc (hunk 12 above)
Comment (lpettyjo) [reply release_notes/ocp-4-18-release-notes.adoc:508]: @radeore It's technically a feature, but it seems more to be addressing something that should have behaved this way all along, so I didn't list it. @gcharot @jsafrane Do you want me to list this in the features table?

File: release_notes/ocp-4-18-release-notes.adoc (hunk 11 above)
Comment (lpettyjo) [reply release_notes/ocp-4-18-release-notes.adoc:503]: Done.

File: release_notes/ocp-4-18-release-notes.adoc [hunk 13]
```diff
@@ -474,9 +473,72 @@ The terminology for the `ip_whitelist` and `ip_blacklist` annotations have been
...
+
+// For information about using the Secrets Store CSI Driver Operator to mount secrets from an external secrets store to a CSI volume, see xref:../nodes/index#nodes-pods-secrets-store[Providing sensitive data to pods by using an external secrets store].
+
+[id="ocp-4-18-release-notes-storage-pv-last-phase-transition-time_{context}"]
```
Comment (RomanBednar) [inline release_notes/ocp-4-18-release-notes.adoc:490]: @lpettyjo lgtm

File: release_notes/ocp-4-18-release-notes.adoc [hunk 14]
```diff
@@ -474,9 +473,72 @@ The terminology for the `ip_whitelist` and `ip_blacklist` annotations have been
 
 [id="ocp-4-18-storage-editing-overprovisioning-ratio_{context}"]
 ==== Over-provisioning ratio update after LVMCluster custom resource creation
-
 Previously, the `thinPoolConfig.overprovisionRatio` field in the `LVMCluster` custom resource (CR) could be configured only during the creation of the `LVMCluster` CR. With this release, you can now update the `thinPoolConfig.overprovisionRatio` field even after creating the `LVMCluster` CR.
 
+[id="ocp-4-18-release-notes-storage-cifs-smb-csi-driver_{context}"]
```
Comment (Phaow) [inline release_notes/ocp-4-18-release-notes.adoc:478]: LGTM for smb csi, we already add the limitations use cases and remove the TP in https://github.com/openshift/openshift-docs/pull/84302

Comment (mpatlasov) [reply release_notes/ocp-4-18-release-notes.adoc:478]: +1. LGTM for smb csi part.

**Comment by dobsonj** [comment]
LGTM :tada:

**Comment by mpatlasov** [comment]
LGTM

File: release_notes/ocp-4-18-release-notes.adoc [hunk 15]
```diff
@@ -490,9 +489,80 @@ For more information, see xref:../networking/ovn_kubernetes_network_provider/ovn
...
+
+For a bound PV and persistent volume claim (PVC) pair, the ordering of PV-PVC deletion determined whether the PV delete reclaim policy was applied or not. The PV applied the reclaim policy if the PVC was deleted prior to deleting the PV. However, if the PV was deleted prior to deleting the PVC, then the reclaim policy was not applied. As a result of that behavior, the associated storage asset in the external infrastructure was not removed.
+
+With {product-title} 4.18, the PV reclaim policy is consistently always applied. This feature has Technical Preview status.
+
+For more information, see xref:../storage/understanding-persistent-storage.adoc#reclaiming_understanding-persistent-storage[Reclaim policy for persistent volumes].
+
+[id="ocp-4-18-release-notes-storage-lso-cleanup_{context}"]
+==== Improved ability to easily remove LVs or LVSs is generally available
+{product-title} 4.18 improves the ability to remove Local Volumes (LVs) and Local Volume Sets (LVSs) by automatically removing artifacts, thus reducing the number of steps required.
```
Comment (gcharot) [inline release_notes/ocp-4-18-release-notes.adoc:530]: I think we should mention context here (i.e LSO)

File: release_notes/ocp-4-18-release-notes.adoc [hunk 16]
```diff
@@ -490,9 +489,80 @@ For more information, see xref:../networking/ovn_kubernetes_network_provider/ovn
...
+
+For more information, see xref:../storage/persistent_storage/persistent_storage_local/persistent-storage-local.adoc#local-removing-device_persistent-storage-local[Removing a local volume or local volume set].
+
+[id="ocp-4-18-release-notes-storage-volume-group-snapshots_{context}"]
+==== CSI volume group snapshots is generally available
+{product-title} 4.18 introduces Container Storage Interface (CSI) volume group snapshots as a generally available feature. CSI volume group snapshots use a label selector to group multiple persistent volume claims (PVCs) for snapshotting. A volume group snapshot represents copies from multiple volumes that are taken at the same point-in-time. This can be useful for applications that contain multiple volumes. 
+
+{product-title} supports volume group snapshots by default. However, the only storage type in {product-title} that supports volume group snapshots is {rh-storage}.
```
Comment (gcharot) [inline release_notes/ocp-4-18-release-notes.adoc:538]: Explicitly mention that this feature is CSI driver dependent. Wording is correct but needs to be rewritten to make sure people understand that the driver needs to implement the feature.

Not sure we should say "only", we have see a case where a third party driver adds supports after 4.18GA. Maybe just say ODF supports it?

**Comment by dfajmon** [comment]
LGTM :1st_place_medal:

File: release_notes/ocp-4-18-release-notes.adoc (hunk 15 above)
Comment (lpettyjo) [reply release_notes/ocp-4-18-release-notes.adoc:530]: Done.

File: release_notes/ocp-4-18-release-notes.adoc (hunk 16 above)
Comment (lpettyjo) [reply release_notes/ocp-4-18-release-notes.adoc:538]: Done.

File: release_notes/ocp-4-18-release-notes.adoc [hunk 17]
```diff
@@ -508,9 +507,80 @@ $ oc patch featuregates cluster -p '{"spec": {"featureSet": "TechPreviewNoUpgrad
...
+
+[id="ocp-4-18-release-notes-storage-lso-cleanup_{context}"]
+==== Improved ability to easily remove LVs or LVSs for LSO is generally available
+For the Local Storage Operator (LSO), {product-title} 4.18 improves the ability to remove Local Volumes (LVs) and Local Volume Sets (LVSs) by automatically removing artifacts, thus reducing the number of steps required.
+
+For more information, see xref:../storage/persistent_storage/persistent_storage_local/persistent-storage-local.adoc#local-removing-device_persistent-storage-local[Removing a local volume or local volume set].
+
+[id="ocp-4-18-release-notes-storage-volume-group-snapshots_{context}"]
+==== CSI volume group snapshots is generally available
+{product-title} 4.18 introduces Container Storage Interface (CSI) volume group snapshots as a generally available feature. The CSI Volume group snapshots feature is implemented by the CSI driver. CSI volume group snapshots use a label selector to group multiple persistent volume claims (PVCs) for snapshotting. A volume group snapshot represents copies from multiple volumes that are taken at the same point-in-time. This can be useful for applications that contain multiple volumes.
```
Comment (gcharot) [inline release_notes/ocp-4-18-release-notes.adoc:554]: > The CSI Volume group snapshots feature is implemented by the CSI driver

It's in OCP for core functionality AND ALSO needs to be supported by the driver. The wording may suggest that this feature entirely relies on the driver which is not the case. Maybe reword it with something like "The feature needs to be supported by the CSI driver"

Comment (lpettyjo) [reply release_notes/ocp-4-18-release-notes.adoc:554]: Done.

**Comment by gcharot** [comment]
LGTM

**Comment by chao007** [comment]
LGTM

**Comment by jsafrane** [comment]
lgtm, however, be careful about the volume group snapshot support. It's not GA in 4.18 yet.

File: release_notes/ocp-4-18-release-notes.adoc (hunk 6 above)
Comment (gcharot) [reply release_notes/ocp-4-18-release-notes.adoc:553]: correct, it may slip GA and added in zstream

Comment (gcharot) [reply release_notes/ocp-4-18-release-notes.adoc:600]: @lpettyjo final decision, this goes tech preview

File: release_notes/ocp-4-18-release-notes.adoc [hunk 18]
```diff
@@ -593,9 +592,80 @@ For more information about the `cluster-compare` plugin, see xref:../scalability
...
+For the Local Storage Operator (LSO), {product-title} 4.18 improves the ability to remove Local Volumes (LVs) and Local Volume Sets (LVSs) by automatically removing artifacts, thus reducing the number of steps required.
+
+For more information, see xref:../storage/persistent_storage/persistent_storage_local/persistent-storage-local.adoc#local-removing-device_persistent-storage-local[Removing a local volume or local volume set].
+
+[id="ocp-4-18-release-notes-storage-volume-group-snapshots_{context}"]
+==== CSI volume group snapshots (Technology Preview)
+{product-title} 4.18 introduces Container Storage Interface (CSI) volume group snapshots as a Technology Preview feature. This feature needs to be supported by the CSI driver. CSI volume group snapshots use a label selector to group multiple persistent volume claims (PVCs) for snapshotting. A volume group snapshot represents copies from multiple volumes that are taken at the same point-in-time. This can be useful for applications that contain multiple volumes. 
+
+{product-title} supports volume group snapshots by default. {rh-storage} supports volume group snapshots.
```
Comment (gcharot) [inline release_notes/ocp-4-18-release-notes.adoc:641]: I think we can remove this line because it's not enabled by default

File: release_notes/ocp-4-18-release-notes.adoc [hunk 19]
```diff
@@ -1754,33 +1828,53 @@ In the following tables, features are marked with the following statuses:
...
 |Migrating CNS Volumes Between Datastores
 |Not Available
-|Not Available
+|Developer Preview
 |Developer Preview
 
+|CSI volume group snapshots
+|Not Available
+|Not Available
+|General Availability
```
Comment (gcharot) [inline release_notes/ocp-4-18-release-notes.adoc:1861]: @lpettyjo we need TP here as well

File: release_notes/ocp-4-18-release-notes.adoc (hunk 6 above)
Comment (lpettyjo) [reply release_notes/ocp-4-18-release-notes.adoc]: Done.

File: release_notes/ocp-4-18-release-notes.adoc (hunk 12 above)
Comment (lpettyjo) [reply release_notes/ocp-4-18-release-notes.adoc:627]: Leaving out of table.

File: release_notes/ocp-4-18-release-notes.adoc (hunk 18 above)
Comment (lpettyjo) [reply release_notes/ocp-4-18-release-notes.adoc:641]: Done.

File: release_notes/ocp-4-18-release-notes.adoc (hunk 19 above)
Comment (lpettyjo) [reply release_notes/ocp-4-18-release-notes.adoc:1861]: Done. Good catch!

**Comment by gcharot** [comment]
LGTM

**Comment by duanwei33** [comment]
LGTM

**Comment by gouthampacha** [comment]
LGTM, thank you!

**Comment by vr4manta** [comment]
LGTM

**Comment by ropatil010** [comment]
LGTM

**Comment by ayushsatyam146** [comment]
LGTM

**Comment by rlobillo** [comment]
LGTM

**Comment by gnufied** [comment]
lgtm
